### PR TITLE
MM-69738 Fix issue with create button disabling on date change

### PR DESCRIPTION
**Pull request by avasconcelos114**
#### Summary

This PR addresses an issue with the create event modal's "Create" button being disabled when a date value changes

This PR also fixes an issue with the linked channel field not being able to be cleared, included in this PR as it was a one-line fix

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-69738
Fixes https://mattermost.atlassian.net/browse/MM-69739

## Change Impact: 🟡 Medium

**Reasoning:** The change adjusts user-facing create-event form validation logic and adds a new datetime helper that’s used by shared time/date constraints (including rounding to a 15-minute step). It also updates the channel selector UX to allow clearing, which impacts the submission payload and validation state.

**Regression Risk:** Medium; while the datetime utility change is targeted and accompanied by unit tests for the new “earliest time today” behavior, it can still affect edge cases around “today” boundaries and time clearing/ordering that determine whether the Create button is enabled.

**QA Recommendation:** Do focused manual QA (moderate amount): create events switching to/from today, changing the date while start/end times are set (especially around the earliest selectable time thresholds), verifying Create button enable/disable, and clearing the linked channel to confirm the submitted value is empty and validation still succeeds. Skipping manual QA carries moderate risk.

*Generated by CodeRabbitAI*

### CODE REVIEW

File: webapp/src/utils/datetime.ts [hunk 1]
```diff
@@ -1,7 +1,19 @@
+const MINUTE_STEP = 15;
```
Comment (nang2049) [inline webapp/src/utils/datetime.ts]: Duplicate of minuteStep in time_selector.tsx. Please export from one place and import into both

File: webapp/src/utils/datetime.ts [hunk 2]
```diff
@@ -1,7 +1,19 @@
...
     const now = new Date();
     const year = now.getFullYear();
     const month = String(now.getMonth() + 1).padStart(2, '0');
     const day = String(now.getDate()).padStart(2, '0');
     return `${year}-${month}-${day}`;
 }
+
+// Earliest selectable time for today, rounded up to the next MINUTE_STEP.
+// Kept consistent with the option constraints in TimeSelector.
+export function getEarliestTimeForToday(): string {
```
Comment (nang2049) [inline webapp/src/utils/datetime.ts:16]: When now is >= 23:45 this returns "24:00". Lex-compare then makes any non empty start_time look "in the past" so the form always clears. Clamp to "23:45" when hour >= 24. Please add a unit test for this edge

**Review by nang2049 — COMMENTED** [review]
LGTM with some suggestions :D

**Review by ogi-m — APPROVED** [review]
Thanks @avasconcelos114 , verified both issues are fixed!

**Review by nang2049 — APPROVED** [review]
